exchanges/okx/okx_test.go

```diff
@@ -0,0 +1,3687 @@
+    {"ordId":"123458","state":"3","ccy": "ETH",      "protocol": "sushiswap","protocolType":"defi",  "term":"0","apy":"0.12","investData":[  {    "ccy":"USDT",    "amt":"100"  },  {    "ccy":"ETH",    "amt":"0.03"  }],"earningData": [ {    "ccy": "SUSHI",    "earningType":"1",    "realizedEarnings":"3" }],"purchasedTime":"1597026383085","redeemedTime":"1597126383085"
+    },{"ordId":"123458","state":"3","ccy": "LON","protocol": "tokenlon","protocolType":"defi","earningCcy": ["LON"],"term":"7","apy":"0.12","investData":[{"ccy":"LON","amt":"1"}],"earningData": [{"ccy": "LON","earningType":"0","realizedEarnings":"3"}],"purchasedTime":"1597026383085","redeemedTime":"1597126383085"}
+]`
+
+func TestGetFundingOrderHistory(t *testing.T) {
+	t.Parallel()
+	var response []FundingOrder
+	if err := json.Unmarshal([]byte(fundingOrderHistoryJSON), &response); err != nil {
+		t.Errorf("%s error while deserializing to FundingOrder %v", ok.Name, err)
+	}
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	if _, err := ok.GetFundingOrderHistory(context.Background(), "", "", "", time.Time{}, time.Time{}, 0); err != nil {
+		t.Errorf("%s GetFundingOrderHistory() error %v", ok.Name, err)
+	}
+}
+
+var systemStatusResponseJSON = `{"title": "Spot System Upgrade","state": "scheduled","begin": "1620723600000","end": "1620724200000","href": "","serviceType": "1","system": "classic","scheDesc": ""}`
+
+func TestSystemStatusResponse(t *testing.T) {
+	t.Parallel()
+	var resp SystemStatusResponse
+	if err := json.Unmarshal([]byte(systemStatusResponseJSON), &resp); err != nil {
+		t.Error("Okx Deserializing to SystemStatusResponse error", err)
+	}
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	if _, err := ok.SystemStatusResponse(context.Background(), ""); err != nil {
+		t.Error("Okx SystemStatusResponse() error", err)
+	}
+}
+
+/**********************************  Wrapper Functions **************************************/
+
+func TestFetchTradablePairs(t *testing.T) {
+	t.Parallel()
+	if _, err := ok.FetchTradablePairs(context.Background(), asset.Futures); err != nil {
+		t.Error("Okx FetchTradablePairs() error", err)
+	}
+}
+
+func TestUpdateTradablePairs(t *testing.T) {
+	t.Parallel()
+	if err := ok.UpdateTradablePairs(context.Background(), true); err != nil {
+		t.Error("Okx UpdateTradablePairs() error", err)
+	}
+}
+
+func TestUpdateTicker(t *testing.T) {
+	t.Parallel()
+	if _, err := ok.UpdateTicker(context.Background(), currency.NewPair(currency.BTC, currency.USDT), asset.Futures); err != nil {
```

Comment:
I don't know what is happening here. We are requesting BTC-USDT and it is fetching a contrac. It seems we are loading in the underlying BTC-USDT which should not be the case. WE are missing loading of these seperate contracts associated with the underlying:

```
{"alias":"this_week","baseCcy":"","category":"1","ctMult":"1","ctType":"linear","ctVal":"0.01","ctValCcy":"BTC","expTime":"1664524800000","instId":"BTC-USDT-220930","instType":"FUTURES","lever":"125","listTime":"1646986200032","lotSz":"1","maxIcebergSz":"1000000","maxLmtSz":"1000000","maxMktSz":"1000","maxStopSz":"1000","maxTriggerSz":"1000000","maxTwapSz":"1000000","minSz":"1","optType":"","quoteCcy":"","settleCcy":"USDT","state":"live","stk":"","tickSz":"0.1","uly":"BTC-USDT"},{"alias":"next_week","baseCcy":"","category":"1","ctMult":"1","ctType":"linear","ctVal":"0.01","ctValCcy":"BTC","expTime":"1665129600000","instId":"BTC-USDT-221007","instType":"FUTURES","lever":"125","listTime":"1663920600039","lotSz":"1","maxIcebergSz":"1000000","maxLmtSz":"1000000","maxMktSz":"1000","maxStopSz":"1000","maxTriggerSz":"1000000","maxTwapSz":"1000000","minSz":"1","optType":"","quoteCcy":"","settleCcy":"USDT","state":"live","stk":"","tickSz":"0.1","uly":"BTC-USDT"},{"alias":"quarter","baseCcy":"
","category":"1","ctMult":"1","ctType":"linear","ctVal":"0.01","ctValCcy":"BTC","expTime":"1672387200000","instId":"BTC-USDT-221230","instType":"FUTURES","lever":"125","listTime":"1654848600669","lotSz":"1","maxIcebergSz":"1000000","maxLmtSz":"1000000","maxMktSz":"1000","maxStopSz":"1000","maxTriggerSz":"1000000","maxTwapSz":"1000000","minSz":"1","optType":"","quoteCcy":"","settleCcy":"USDT","state":"live","stk":"","tickSz":"0.1","uly":"BTC-USDT"},{"alias":"next_quarter","baseCcy":"","category":"1","ctMult":"1","ctType":"linear","ctVal":"0.01","ctValCcy":"BTC","expTime":"1680249600000","instId":"BTC-USDT-230331","instType":"FUTURES","lever":"125","listTime":"1663315800000","lotSz":"1","maxIcebergSz":"1000000","maxLmtSz":"1000000","maxMktSz":"1000","maxStopSz":"1000","maxTriggerSz":"1000000","maxTwapSz":"1000000","minSz":"1","optType":"","quoteCcy":"","settleCcy":"USDT","state":"live","stk":"","tickSz":"0.1","uly":"BTC-USDT"}
```

I Should be expecting this endpoint to fail if BTC-USDT is requested for futures and should be specifically set as currency.NewPairFromStrings("BTC", "USDT-220930")